exporters/user_events/CMakeLists.txt

```diff
@@ -26,10 +27,12 @@ if(MAIN_PROJECT)
   find_package(opentelemetry-cpp REQUIRED)
 endif()
 
-# don't build samples and tools from LinuxTracepoints.
-set(BUILD_SAMPLES OFF)
-set(BUILD_TOOLS OFF)
-add_subdirectory(third_party/LinuxTracepoints)
+if(BUILD_TRACEPOINTS)
+  # don't build samples and tools from LinuxTracepoints.
+  set(BUILD_SAMPLES OFF)
+  set(BUILD_TOOLS OFF)
```

Comment:
Restore the potential value in `BUILD_SAMPLES` and `BUILD_TOOLS` after the call to `add_subdirectory()`? This was missed before, but I think it will be great to have it.

Reply:
I've added the usage of [block()](https://cmake.org/cmake/help/latest/command/block.html) which makes this automatic. Nonetheless, this is added after CMake 3.25, and the CMake file was using 3.12. If updating the CMake version is a blocker, I can reverse this and try a more manual approach storing the values in some variables.

Otherwise, we can keep block() which is a bit cleaner.

Reply:
Seems this version of CMake is not available in the Github CI runner by default.
>  CMake 3.25 or higher is required.  You are running version 3.16.3

Reply:
Makes sense, I've restored CMake version to 3.12 and just use some variables to store the values before adding the subdirectory. Thanks!